public/components/palantir/index.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './index.scss';
+
+import {
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiPopoverTitle,
+  EuiText,
+  EuiPopoverFooter,
+  EuiBadge,
+} from '@elastic/eui';
+import React, { Children, isValidElement, useState } from 'react';
+import { Palantir as PalantirInput } from '../../types';
+import { getPalantirRegistry } from '../../services';
+
+// TODO: type content
+// ?? what is the content going to be? does it change from anchor to anchor
+// ?? what is the anchor point? will assume character based
+export const Palantir = ({
+  children,
+  key,
+  input,
+  onSubmit,
+}: {
+  children: React.ReactNode;
+  key?: string;
+  input?: PalantirInput;
+  onSubmit?: () => void;
+}) => {
+  const registry = getPalantirRegistry();
+  const [isVisible, setIsVisible] = useState(false);
+
+  // TODO: error handling for bad implemention from plugins or just don't render?
+  const onAnchorClick = () => setIsVisible((visible) => !visible);
+  const onSubmitClick = async (palantirInput: PalantirInput) => {
+    setIsVisible(false);
+    if (onSubmit) {
+      onSubmit();
+      return;
+    }
+
+    // TODO: we have access to the React child. We can crawl it for data and values to pass for more context
+    // Example: We can get the text and provide the chat bot more information or think about supporting tokens in the registry
+    registry.open(palantirInput);
+  };
+
+  const closeAssistantPopover = () => setIsVisible(false);
+  const anchorContent = (
+    <EuiButtonEmpty
+      className="palantirAnchorButton"
+      size="xs"
+      flush="left"
+      iconType="chatRight"
+      iconSide="left"
+      onClick={onAnchorClick}
+    >
+      <span className="palantirAnchorContent">{children}</span>
+    </EuiButtonEmpty>
+  );
+
+  // TODO: i18n && delay
+  // TODO: compressed loading component (empty prompt)
+  // TODO: --- loading 2-3 seconds
+  const getPalantirInputByChildKey = () => {
+    let palantir;
+    Children.forEach(children, (child) => {
+      if (isValidElement(child) && child.key && registry.get(child.key as string)) {
+        palantir = registry.get(child.key as string);
+      }
+    });
+    if (!palantir) throw Error('Child key not found in registry.');
+    return palantir;
+  };
+
+  const palantirInput = input ?? getPalantirInputByChildKey();
+  const assistantPopover: React.ReactNode = (
+    <EuiPopover
+      key={key ?? palantirInput.key}
+      button={anchorContent}
+      isOpen={isVisible}
+      closePopover={closeAssistantPopover}
+      anchorClassName="palantirAnchor"
+      anchorPosition="upCenter"
+      display="block"
+    >
+      <EuiPopoverTitle>{children}</EuiPopoverTitle>
```

Comment:
Q: Why render `children` again in popover title?

Reply:
i can sanitize this as the HTML is still coming through but I believe it was to the original UX suggestion. but it has been a while.

cc: @opensearch-project/opensearch-ux

Reply:
I see, I was thinking the `children` could be something else other than a simple text node, if that's the case, the popover may look a bit strange?

Reply:
yeah. have updated it. also i think there were mocks at one point that add visualizations so its prolly a little bit over kill with the updated mocks i have modify it a bit

Reply:
here is out it currently looks. still not in its final form but without the children:

default:
<img width="272" alt="Screenshot 2023-12-21 at 7 40 10 AM" src="https://github.com/opensearch-project/dashboards-assistant/assets/12060704/844df0ff-7555-4716-a8e1-5065f3061938">

hover:
<img width="235" alt="Screenshot 2023-12-21 at 7 40 26 AM" src="https://github.com/opensearch-project/dashboards-assistant/assets/12060704/14fd2280-70c8-47aa-8c58-f9e6e17b7662">

clicked:
<img width="251" alt="Screenshot 2023-12-21 at 7 40 37 AM" src="https://github.com/opensearch-project/dashboards-assistant/assets/12060704/42c6b98d-dc42-41cf-b527-e1909d4f181d">

still need to keep improving it.